agents/model_capabilities.py

```
@@ -0,0 +1,197 @@
+    if "/" in model_name:
+        return model_name
+    return f"{_OPENROUTER_PREFIX.get(provider, provider)}/{model_name}"
+
+
+def _load(source: str) -> dict:
+    path = _CACHE_DIR / f"{source}.json"
+    if path.exists() and time.time() - path.stat().st_mtime < _TTL_SECONDS:
+        return json.loads(path.read_text())
+    try:
+        # Why: models.dev rejects urllib's default UA with 403.
+        req = urllib.request.Request(_SOURCES[source], headers={"User-Agent": "codeboarding/1.0"})
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)